counter_robots/robot_detection.py

```diff
@@ -0,0 +1,37 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of COUNTER-Robots.
+# Copyright (C) 2018 CERN.
+#
+# COUNTER-Robots is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+
+"""Robot and machine detection."""
+
+import re
+
+from counter_robots.config import config
+
+
+def is_robot(user_agent):
+    """Detect if the user agent belongs to a robot."""
+    if user_agent is None:
+        return False
+    robots_regexp = _regexp(config['ROBOTS_FILE'])
+    return bool(robots_regexp.search(user_agent))
+
+
+def is_machine(user_agent):
+    """Detect if a user agent belongs to a machine."""
+    if user_agent is None:
+        return True
+    machines_regexp = _regexp(config['MACHINES_FILE'])
+    return bool(machines_regexp.search(user_agent))
+
+
+def _regexp(filename):
```

Comment:
memoize/cache this like it's done in https://github.com/inveniosoftware/invenio-collections/blob/b120e1461884a79afb994be4c95703a02edb578d/invenio_collections/receivers.py#L29-L32